libs/providers/flagd-web/src/lib/flagd-web-provider.ts

```diff
@@ -0,0 +1,304 @@
+    const transport = createConnectTransport({
+      baseUrl: `${tls ? "https" : "http"}://${host}:${port}`
+    });
+    this.promiseClient = createPromiseClient(Service, transport);
+    this.callbackClient = createCallbackClient(Service, transport);
+    this.cache = new NodeCache({
+      stdTTL: cacheTTL
+    })
+    if (cacheMaxBytes != 0) {
+      this.cache.addListener("set", () => {
+        const s = this.cache.getStats()
+        if (s.ksize + s.vsize > cacheMaxBytes) {
+          console.log(`maximum cache size of ${cacheMaxBytes} reached: busting cache`)
+          this.cache.flushAll();
+        }
+      })
+    }
+    this.cacheActive = cache
+    this.initStream(0, maxRetries)
+
+  }
+
+  async initStream(i: number, maxRetries: number) {
+    this.callbackClient.eventStream(
+      {},
+      (message) => {
+        console.log(`event received: ${message.type}`);
+        switch (message.type) {
+          case EVENT_PROVIDER_READY:
+            this.providerReady = true;
+            return
+          case EVENT_CONFIGURATION_CHANGE: {
+            const s = message.data?.toJsonString()
+            if (s === undefined) {
+              console.error("configuration change: malformed configuration change received, flushing all", message)
+              this.cache.flushAll()
+              return
+            }
+            const body = JSON.parse(s) as ConfigurationChangeBody
+            const keys = this.cache.keys()
+            let deleted = 0
+            for (let i=0;i<keys.length;i++) {
+              const prefix = keys[i].split("--")[0]
+              if (body.flagKey === prefix) {
+                deleted ++
+                this.cache.del(keys[i])
+              }
+            }
+            console.log(`configuration change: ${body.type}  ${body.flagKey}  ${body.source} cache items busted: ${deleted}`)
+            return
+          }
+        }
+      },
+      async () => {
+        if (i != maxRetries) {
+          const delay = (i+1) * Math.random() * 300
+          console.log(`connection failed on attempt ${i+1}, retrying in ${delay}ms`)
+          await new Promise(f => setTimeout(f, delay));
+          this.initStream(i+1, maxRetries)
+        } else {
+          console.log("could not establish connection to flagd")
+          this.connectionError = true
+        }
+      },
+    )
+  }
+
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    transformedContext: EvaluationContext
+  ): Promise<ResolutionDetails<boolean>> {
+    if (!this.providerReady) {
+      return Promise.resolve({
+        value: defaultValue,
+        reason: StandardResolutionReasons.ERROR,
+        errorCode: this.connectionError ? ErrorCode.GENERAL : ErrorCode.PROVIDER_NOT_READY
+      })
+    }
+    const req = {
+      flagKey,
+      context: Struct.fromJsonString(JSON.stringify(transformedContext)),
+    }
+    const reqHash = `${flagKey}--${Md5.hashStr(JSON.stringify(req.context))}`;
```

Comment:
This seems like a separate method would be useful, this is repeated in every resolve method.

Reply:
Ya, in https://github.com/open-feature/js-sdk-contrib/pull/112 these were all separate methods. One of my goals with this work was to remove some of the duplication with generics, that work I've done.